spec/features/admin/bulk_product_update_spec.rb

```diff
@@ -516,11 +516,16 @@
       it "shows an edit button for products, which takes the user to the standard edit page for that product" do
         expect(page).to have_selector "a.edit-product", count: 2
 
-        within "tr#p_#{p1.id}" do
-          find("a.edit-product").click
+        new_window = window_opened_by do 
+          within "tr#p_#{p1.id}" do
+            find("a.edit-product").click
+          end
         end
```

Comment:
nice trick!
just a thought: is this really necessary?
if the expect URI is being evaluated too early, the usual solution in OFN's code base is to expect some DOM element in the destination page before the required verification, in this case the URL, is done.

Reply:
I believe that is necessary, since the feature request was to open a new window.

I suppose we could examine the DOM on the existing page for the presence of a _blank target on the appropriate link, but I think checking for the existence of the new window is a truer test for covering the required behaviour. Although I'm happy to implement either case (or another you can think of) -- you're call.

I do need to change the description of the test, though. I realise I kept the old one, whilst changing what the test looked for. :man_facepalming:

Reply:
ok, I missed the fact the request is to open a new window, thus the test. It's perfect!